vllm/v1/spec_decode/eagle.py

```diff
@@ -103,22 +108,52 @@ def propose(
             # Update the inputs.
             input_ids = draft_token_ids_list[-1]
             positions += 1
+
+            # NOTE(woosuk): We should handle the case where the draft model
+            # generates tokens beyond the max model length. Since it is complex
+            # to remove such requests from the batch, we keep them in the batch
+            # but adjust the position ids and slot mappings to avoid the
+            # out-of-range access during the model execution. The draft tokens
+            # generated with this adjustment should be ignored.
```

Comment:
Do we need to add the logic to ignoring such draft tokens [here](https://github.com/vllm-project/vllm/pull/16087/files#diff-80ee7e2a62f9dcfbb8a312dc4e3948557e97ef187290daebbcae1e28596bda29R1200)?

Reply:
Good question. The scheduler handles it https://github.com/vllm-project/vllm/blob/af7462b6d1ebc10ca83da57f2804c9e49a40e83c/vllm/v1/core/sched/scheduler.py#L188-L193